src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.OperationVisitor.cs

```diff
@@ -36,9 +37,19 @@ private sealed class OperationVisitor : GlobalFlowStateDataFlowOperationVisitor
 
                 if (_platformCheckMethods.Contains(method.OriginalDefinition))
                 {
-                    return PlatformMethodValue.TryDecode(method, visitedArguments, DataFlowAnalysisContext.ValueContentAnalysisResult, _osPlatformType, out var platformInfo) ?
-                        new GlobalFlowStateAnalysisValueSet(platformInfo) :
-                        GlobalFlowStateAnalysisValueSet.Unknown;
+                    using var infosBuilder = ArrayBuilder<PlatformMethodValue>.GetInstance();
+                    if (PlatformMethodValue.TryDecode(method, visitedArguments, DataFlowAnalysisContext.ValueContentAnalysisResult, _osPlatformType, infosBuilder))
```

Comment:
TryDecode can return multiple platform infos now via dataflow. Unit test has been added for this case.

Reply:
> TryDecode can return multiple platform infos now via dataflow

Just curious why is that? Even there were multiple local variables only one can be used with the method, just not sure what exact case make it need to use/return multiple

Reply:
There is a test I added for that scenario. Even though there is a single argument, it can point to multiple potential values at runtime as assignments can be conditional

```csharp
        var windowsPlatform = OSPlatform.Create(""Windows"");
        var linuxPlatform = OSPlatform.Create(""Linux"");
        var platform = isWindows ? windowsPlatform : linuxPlatform;
        if(RuntimeInformation.IsOSPlatform(platform)) // Value here is "Windows" or "Linux"
        {
            M2();
        }
        else
        {
            {|#0:M2()|};
        }
```

Reply:
Ah i see, i saw that test but thought it would be clear if the value will be Windows or Linux, forgot we are doing static analysis 😛. Thanks for explanation!